aehmc/metrics.py

```diff
@@ -12,7 +12,7 @@ def gaussian_metric(
     shape = aet.shape(inverse_mass_matrix)[0]
```

Comment:
This is the start of where the shape/broadcast information is being lost.

In your test, `inverse_mass_matrix.broadcastable == [True]`, which means that `inverse_mass_matrix.shape == (1,)`.  `aet.shape(inverse_mass_matrix)[0]` will not directly return one, though, so, when `shape` is used later as the `size` parameter for `norm_samples` below, the `RandomVariable` `Op` isn't able to determine that `norm_samples` is also broadcastable.  To the `RandomVariable` `Op`, `shape` could be anything, so it sets `norm_samples.broadcastable` to `[False]` just to be "safe".

I changed this line locally to use `aesara.tensor.shape.shape_tuple` instead, and that will make `shape` a `ScalarConstant` of `1`; however, it looks like there might be an issue with the `RandomVariable` `Op`, because `norm_samples.broadcastable` is still `[False]`.
`norm_samples` below will be given a shape

Reply:
```suggestion
    shape = shape_tuple(inverse_mass_matrix)[0]
```